### ISSUE TITLE
Automated submission of issues from external CI

### ISSUE THREAD

**Issue by ejgallego** [rocq-prover/bot#110]
In relation to https://github.com/coq/coq/issues/10418 and some discussion on the Coq call, it would be very nice if coq.io could submit an issue automatically to Coq's opam repository when it detects a problem. [Even a PR could be done fixing the issue, or maybe coqbot could propose a fix that could be automatically done as `coqbot: implement suggestion`

cc: @palmskog @clarus

**Comment by Zimmi48** [comment]
Sure, that would be very easy to do. I could add a specific endpoint, to which coq.io would send the text to put in the new issue. But is there a risk of posting duplicate issues? And I would need more explanations to understand the case of PRs.

**Comment by ejgallego** [comment]
> But is there a risk of posting duplicate issues?

Indeed that's a very good point, checking that would require some amount of state.

> And I would need more explanations to understand the case of PRs.

For the PRs I was thinking that maybe coqbot could have some logic as to for example add lower/upper bounds to opam definitions, etc... just a thought on automatic repos repair.

**Comment by Zimmi48** [comment]
> > But is there a risk of posting duplicate issues?
> 
> Indeed that's a very good point, checking that would require some amount of state.

Or the bot could check the list of open issues for some identifier.

**Comment by ejgallego** [comment]
> Or the bot could check the list of open issues for some identifier.

Indeed that could work, but wouldn't it be quite expensive in terms of API calls ? [unless you have some cache tho]

**Comment by Zimmi48** [comment]
No, a single request should be enough. The GraphQL API is pretty powerful. For instance, with this request I can see all issues opened by myself on the Coq repository:

```graphql
query { 
  repository(name:"coq",owner:"coq") {
    issues(filterBy:{createdBy:"Zimmi48", states:[OPEN]},first:100) {
      nodes {
        number
        title
        bodyText
      }
      totalCount
    }
  }
}
```

You can test this at https://developer.github.com/v4/explorer/ if you want.

**Comment by clarus** [comment]
For submitting a PR, I think this is quite hard, because when there is an error this could be related to an update of any of the package dependencies. Plus there are sometimes false positives (timeouts, ...). For me it is fine to do PRs by hand for now.

For me the Zulip chat is fine for now to report the issues.

We can also watch what people do in the OCaml community to maintain the repository. I do not think they have GitHub bots for issues / pull-requests for now, but may be mistaken.

**Comment by Zimmi48** [comment]
The OCaml community has often been lagging behind us if you want my opinion 😉
That being said, if you think there is no need for more automation for now, then no need to overthink this.
